lib/index.js

```diff
@@ -23,11 +23,11 @@ exports.clone = function (obj, seen) {
         return obj;
     }
 
-    seen = seen || { orig: [], copy: [] };
+    seen = seen || new Map();
 
-    const lookup = seen.orig.indexOf(obj);
-    if (lookup !== -1) {
-        return seen.copy[lookup];
+    const lookup = seen.has(obj);
```

Comment:
I was worried something would be falsy in the future, javascript being what it is, but we'll see.